File: src/main/java/kr/hyfata/zero/gui/InventoryGUI.java
package kr.hyfata.zero.gui;

import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;

public interface InventoryGUI {
    void openInventory(Player p);
    void inventoryClickEvent(InventoryClickEvent e);
    void inventoryCloseEvent(InventoryCloseEvent e);
    boolean contains(org.bukkit.inventory.Inventory inventory);
}

File: src/main/java/kr/hyfata/zero/gui/InventoryGUI.kt
package kr.hyfata.zero.gui

import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.inventory.Inventory

interface InventoryGUI {
    fun openInventory(p: Player)
    fun inventoryClickEvent(e: InventoryClickEvent)
    fun inventoryCloseEvent(e: InventoryCloseEvent)
    fun closeInventoryAllPlayers()
    fun contains(inventory: Inventory): Boolean
}